azvault/azvault.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2023, 2025, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package azvault
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/prometheus/common/promslog"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/keyvault/azsecrets"
+)
+
+func GetVaultSecret(vaultId string, secretName string) string {
+	promLogConfig := &promslog.Config{}
+	logger := promslog.New(promLogConfig)
+
+	vaultURI := fmt.Sprintf("https://%s.vault.azure.net/", vaultId)
+
+	// create a credential
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		logger.Error("Failed to obtain an Azure Credential", "err", err)
```

Comment:
Is this not a fatal error? Why do we log the error and continue on

Reply:
Looks like the ocivault code treats any errors as fatal errors: helpers.FatalIfError(err)